kernel/src/hil/screen.rs

```diff
@@ -212,7 +273,16 @@ pub trait Screen {
         height: usize,
     ) -> Result<(), ErrorCode>;
 
-    /// Sends a write command to write data in the selected video memory frame.
+    /// Writes data into the selected rectangle of pixels.
+    ///
+    /// By default, the selected rectangle covers the entire frame.
+    /// The rectangle can be set using `set_write_frame`.
+    ///
+    /// The pixel data contained in the buffer must adhere
+    /// to the current pixel format and grid shape. See `get_pixel_format`.
+    /// Consecutive pixels on the screen will be set
+    /// to consecutive values from the buffer. See `PixelStreamFormat`.
+    ///
```

Comment:
Generally, you should operate on a frame buffer, then apply a rotation after. Otherwise, you have to be continually transforming. I.e., always operate in the framebuffer frame of reference, not the screen frame of reference.

Reply:
Should that interpretation find itself into this change? I deliberately ignored it to limit the scope of this, and we have more semantics questions (around resolution, format, inversion) to sort out.